Rename References to Useful Resources

robmoss opened the pull request:
I've reorganised the sub-headings and content, and added resources from our most recent CoP meeting (2023-08-15).

If this list becomes much larger, we may need to split it out into separate pages. But I think it's okay a single page for now.

robmoss commented:
That's what I get for not checking the build output on my laptop: the original commit left a broken link to the old page in the Collaborating section. Thank you CI for catching the error and letting me know.

EamonConway commented:
This works locally on my machine and I cannot find any broken links to the references page.

robmoss committed:
Rename References to Useful Resources

I've reorganised the sub-headings and content, and added resources from
one of our CoP meetings (2023-08-15).
